src/vips/Image_methods.lua

```diff
@@ -185,18 +185,20 @@ function Image.new_from_array(array, scale, offset)
     local height = #array
 
     local n = width * height
-    local a = ffi.new(gvalue.pdouble_typeof, n)
+    local a = {}
     for y = 0, height - 1 do
         for x = 0, width - 1 do
             a[x + y * width] = array[y + 1][x + 1]
         end
     end
+    a = ffi.new(gvalue.double_arr_typeof, n, a)
```

Comment:
Oooof, changing the type of a is a bit ugly. I'll adjust this.

Reply:
Hopefully fixed with https://github.com/libvips/lua-vips/pull/29/commits/d2c062f7a5d360f68810b3f894760a14ee83f46f.